src/exercises/day-1/luhn.rs

```diff
@@ -15,34 +15,38 @@
 // ANCHOR: solution
 // ANCHOR: luhn
 pub fn luhn(cc_number: &str) -> bool {
-    // ANCHOR_END: luhn
+    let cc_number = cc_number.replace(" ", "");
     let mut sum = 0;
-    let mut double = false;
-    let mut digit_seen = 0;
+    let mut even = false;
 
-    for c in cc_number.chars().filter(|&f| f != ' ').rev() {
-        if let Some(digit) = c.to_digit(10) {
-            if double {
-                let double_digit = digit * 2;
-                sum += if double_digit > 9 {
-                    double_digit - 9
-                } else {
-                    double_digit
-                };
-            } else {
-                sum += digit;
-            }
-            double = !double;
-            digit_seen += 1;
-        } else {
-            return false;
-        }
+    if let Result::Err(_) = cc_number.parse::<u128>() {
+        return false;
     }
 
-    if digit_seen < 2 {
+    if cc_number.len() < 2 {
         return false;
     }
 
+    for i in (0..cc_number.len()).rev() {
+        let mut num = cc_number
+            .chars() //for new line
+            .nth(i)
+            .unwrap()
+            .to_digit(10)
+            .unwrap();
```

Comment:
The use of `unwrap` here is a step back: the code here now relies on the `parse` call above to not error out. It is better to handle this via the code structure (e.g., `match` or similar).